File: web-sdk/src/data/services/yalo-media/yalo-media-service.ts
// Copyright (c) Yalochat, Inc. All rights reserved.

import type { Result } from '@domain/common/result';

export interface MediaUploadResponse {
  id: string;
  signed_url: string;
  original_name: string;
  type: string;
  metadata: Record<string, unknown>;
  created_at: string;
  expires_at: string;
}

export interface YaloMediaService {
  uploadMedia(file: File): Promise<Result<MediaUploadResponse>>;
  downloadMedia(url: string): Promise<Result<Blob>>;
}

File: flutter-sdk/lib/src/data/services/yalo_media/yalo_media_service.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'dart:typed_data';

import 'package:yalo_chat_flutter_sdk/src/common/result.dart';
import 'package:yalo_chat_flutter_sdk/src/data/services/yalo_media/media_upload_response.dart';
import 'package:image_picker/image_picker.dart';

abstract class YaloMediaService {
  Future<Result<MediaUploadResponse>> uploadMedia(XFile file);
  Future<Result<Uint8List>> downloadMedia(String url);
}
